### ISSUE TITLE
Does this work with theme "just-the-docs"?

### ISSUE THREAD

**Issue by emorrow38** [BurdetteLamar/MarkdownHelper#129]
Burdette,

I really want to use markdown_helper with my planned website using "just-the-docs".  To be able to include common markdown text would be a huge advantage.

Unfortunately I can't seem to get it to work (perhaps my fault).

I have a simple folder that is part of a "just-the-docs" website.  The website uses the "docs" folder and under that we have folder "Basic Views Training".    That folder has two files only:

<img width="260" alt="Folder 01 Contents 02" src="https://user-images.githubusercontent.com/68985134/171565940-97da3a34-f3cd-46fe-849c-687f8fb17462.png">

The file Common 01.md is the text to be included and it is referenced from Test Include.md.

Common 01.md looks like this:

<img width="347" alt="Topic Common 01 01 VSCode 01" src="https://user-images.githubusercontent.com/68985134/171566191-3161f614-5ae2-4543-97e3-317a9016a945.png">

Test Include.md looks like this:

<img width="266" alt="Topic Test Include 02 VSCode 01" src="https://user-images.githubusercontent.com/68985134/171566275-8dd03b38-e470-48d4-8367-5de441468203.png">

When I do a Jekyll serve to test this locally the include does not happen:

<img width="388" alt="Website 01 No include 01" src="https://user-images.githubusercontent.com/68985134/171566379-4fb69228-5c44-4b39-b3ba-c998152f9ad9.png">

If I click on the text "markdown" in the Test Include file, it takes me to the Common 01 file, but I didn't want a link - I wanted the text included there.  

I really want this to work - it will be fabulous for my website if it did.

Any ideas on what I can do?

Eugene Morrow, Australia.  (I will probably reply to you overnight your time if you reply to me.)

**Comment by emorrow38** [comment]
Thought I should give you more information.

It occurs to me that my symptoms suggest that markdown_helper is not installed.   I attach a screenshot of the install of markdown_helper:

<img width="366" alt="MH 01 Install 01" src="https://user-images.githubusercontent.com/68985134/171612001-85fbf009-f926-42d5-a08a-3b7b766114f4.png">

I also list the Gemfile.lock for my Jekyll just-the-docs website (although markdown_helper does not appear):

GEM
  remote: https://rubygems.org/
  specs:
    activesupport (3.2.22.5)
      i18n (~> 0.6, >= 0.6.4)
      multi_json (~> 1.0)
    addressable (2.8.0)
      public_suffix (>= 2.0.2, < 5.0)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.11.1)
    colorator (1.1.0)
    commonmarker (0.17.13)
      ruby-enum (~> 0.5)
    concurrent-ruby (1.1.9)
    dnsruby (1.61.7)
      simpleidn (~> 0.1)
    em-websocket (0.5.2)
      eventmachine (>= 0.12.9)
      http_parser.rb (~> 0.6.0)
    ethon (0.14.0)
      ffi (>= 1.15.0)
    eventmachine (1.2.7)
    execjs (2.8.1)
    faraday (1.7.0)
      faraday-em_http (~> 1.0)
      faraday-em_synchrony (~> 1.0)
      faraday-excon (~> 1.1)
      faraday-httpclient (~> 1.0.1)
      faraday-net_http (~> 1.0)
      faraday-net_http_persistent (~> 1.1)
      faraday-patron (~> 1.0)
      faraday-rack (~> 1.0)
      multipart-post (>= 1.2, < 3)
      ruby2_keywords (>= 0.0.4)
    faraday-em_http (1.0.0)
    faraday-em_synchrony (1.0.0)
    faraday-excon (1.1.0)
    faraday-httpclient (1.0.1)
    faraday-net_http (1.0.1)
    faraday-net_http_persistent (1.2.0)
    faraday-patron (1.0.0)
    faraday-rack (1.0.0)
    ffi (1.15.3)
    forwardable-extended (2.6.0)
    gemoji (3.0.1)
    github-pages (217)
      github-pages-health-check (= 1.17.2)
      jekyll (= 3.9.0)
      jekyll-avatar (= 0.7.0)
      jekyll-coffeescript (= 1.1.1)
      jekyll-commonmark-ghpages (= 0.1.6)
      jekyll-default-layout (= 0.1.4)
      jekyll-feed (= 0.15.1)
      jekyll-gist (= 1.5.0)
      jekyll-github-metadata (= 2.13.0)
      jekyll-mentions (= 1.6.0)
      jekyll-optional-front-matter (= 0.3.2)
      jekyll-paginate (= 1.1.0)
      jekyll-readme-index (= 0.3.0)
      jekyll-redirect-from (= 0.16.0)
      jekyll-relative-links (= 0.6.1)
      jekyll-remote-theme (= 0.4.3)
      jekyll-sass-converter (= 1.5.2)
      jekyll-seo-tag (= 2.7.1)
      jekyll-sitemap (= 1.4.0)
      jekyll-titles-from-headings (= 0.5.3)
      jemoji (= 0.12.0)
      kramdown (= 2.3.1)
      kramdown-parser-gfm (= 1.1.0)
      liquid (= 4.0.3)
      mercenary (~> 0.3)
      minima (= 2.5.1)
      nokogiri (>= 1.10.4, < 2.0)
      rouge (= 3.26.0)
      terminal-table (~> 1.4)
    github-pages-health-check (1.17.2)
      addressable (~> 2.3)
      dnsruby (~> 1.60)
      octokit (~> 4.0)
      public_suffix (>= 2.0.2, < 5.0)
      typhoeus (~> 1.3)
    html-pipeline (2.14.0)
      activesupport (>= 2)
      nokogiri (>= 1.4)
    http_parser.rb (0.6.0)
    i18n (0.9.5)
      concurrent-ruby (~> 1.0)
    jekyll (3.9.0)
      addressable (~> 2.4)
      colorator (~> 1.0)
      em-websocket (~> 0.5)
      i18n (~> 0.7)
      jekyll-sass-converter (~> 1.0)
      jekyll-watch (~> 2.0)
      kramdown (>= 1.17, < 3)
      liquid (~> 4.0)
      mercenary (~> 0.3.3)
      pathutil (~> 0.9)
      rouge (>= 1.7, < 4)
      safe_yaml (~> 1.0)
    jekyll-avatar (0.7.0)
      jekyll (>= 3.0, < 5.0)
    jekyll-coffeescript (1.1.1)
      coffee-script (~> 2.2)
      coffee-script-source (~> 1.11.1)
    jekyll-commonmark (1.3.1)
      commonmarker (~> 0.14)
      jekyll (>= 3.7, < 5.0)
    jekyll-commonmark-ghpages (0.1.6)
      commonmarker (~> 0.17.6)
      jekyll-commonmark (~> 1.2)
      rouge (>= 2.0, < 4.0)
    jekyll-default-layout (0.1.4)
      jekyll (~> 3.0)
    jekyll-feed (0.15.1)
      jekyll (>= 3.7, < 5.0)
    jekyll-gist (1.5.0)
      octokit (~> 4.2)
    jekyll-github-metadata (2.13.0)
      jekyll (>= 3.4, < 5.0)
      octokit (~> 4.0, != 4.4.0)
    jekyll-mentions (1.6.0)
      html-pipeline (~> 2.3)
      jekyll (>= 3.7, < 5.0)
    jekyll-optional-front-matter (0.3.2)
      jekyll (>= 3.0, < 5.0)
    jekyll-paginate (1.1.0)
    jekyll-readme-index (0.3.0)
      jekyll (>= 3.0, < 5.0)
    jekyll-redirect-from (0.16.0)
      jekyll (>= 3.3, < 5.0)
    jekyll-relative-links (0.6.1)
      jekyll (>= 3.3, < 5.0)
    jekyll-remote-theme (0.4.3)
      addressable (~> 2.0)
      jekyll (>= 3.5, < 5.0)
      jekyll-sass-converter (>= 1.0, <= 3.0.0, != 2.0.0)
      rubyzip (>= 1.3.0, < 3.0)
    jekyll-sass-converter (1.5.2)
      sass (~> 3.4)
    jekyll-seo-tag (2.7.1)
      jekyll (>= 3.8, < 5.0)
    jekyll-sitemap (1.4.0)
      jekyll (>= 3.7, < 5.0)
    jekyll-titles-from-headings (0.5.3)
      jekyll (>= 3.3, < 5.0)
    jekyll-watch (2.2.1)
      listen (~> 3.0)
    jemoji (0.12.0)
      gemoji (~> 3.0)
      html-pipeline (~> 2.2)
      jekyll (>= 3.0, < 5.0)
    just-the-docs (0.3.3)
      jekyll (>= 3.8.5)
      jekyll-seo-tag (~> 2.0)
      rake (>= 12.3.1, < 13.1.0)
    kramdown (2.3.1)
      rexml
    kramdown-parser-gfm (1.1.0)
      kramdown (~> 2.0)
    liquid (4.0.3)
    listen (3.7.0)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    mercenary (0.3.6)
    mini_portile2 (2.8.0)
    minima (2.5.1)
      jekyll (>= 3.5, < 5.0)
      jekyll-feed (~> 0.9)
      jekyll-seo-tag (~> 2.1)
    multi_json (1.15.0)
    multipart-post (2.1.1)
    nokogiri (1.13.3)
      mini_portile2 (~> 2.8.0)
      racc (~> 1.4)
    octokit (4.21.0)
      faraday (>= 0.9)
      sawyer (~> 0.8.0, >= 0.5.3)
    pathutil (0.16.2)
      forwardable-extended (~> 2.6)
    public_suffix (4.0.6)
    racc (1.6.0)
    rake (13.0.6)
    rb-fsevent (0.11.0)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    rexml (3.2.5)
    rouge (3.26.0)
    ruby-enum (0.9.0)
      i18n
    ruby2_keywords (0.0.5)
    rubyzip (2.3.2)
    safe_yaml (1.0.5)
    sass (3.7.4)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sawyer (0.8.2)
      addressable (>= 2.3.5)
      faraday (> 0.8, < 2.0)
    simpleidn (0.2.1)
      unf (~> 0.1.4)
    terminal-table (1.8.0)
      unicode-display_width (~> 1.1, >= 1.1.1)
    typhoeus (1.4.0)
      ethon (>= 0.9.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.7)
    unicode-display_width (1.7.0)
    wdm (0.1.1)
    webrick (1.7.0)

PLATFORMS
  x64-mingw-ucrt
  x64-mingw32
  x86_64-linux

DEPENDENCIES
  github-pages (~> 217)
  jekyll-feed (~> 0.12)
  just-the-docs
  tzinfo (~> 1.0.0)
  tzinfo-data
  wdm (>= 0.1.0)
  webrick

BUNDLED WITH
   2.2.26

**Comment by BurdetteLamar** [comment]
Don't be offended by the "is it plugged in?" equivalent question:  It's a pre-processor, so "did you run the markdown helper program?"

**Comment by emorrow38** [comment]
Burdette,

Excellent question - I have solved the riddle.

I am tech writer, and I have previously worked in DITA - where what i did in my first posts is enough to get includes working.    In my enthusiasm for markdown_helper I did not take in that I need to run these "markdown_helper" commands and then the Jekyll serve.  I have successfully run markdown_helper and understand it much better now.

CONCLUSIONS:
1.  Yes, markdown_helper works fine with Jekyll theme "just-the-docs" and with any other theme because markdown_helper works on markdown itself and doesn't care how you arrange the expanded markdown files.
1.  I will need to think a lot about how I use markdown_helper.  Our website will be around 1,000 topics with something like 100 includes.  I will need to work out logistics so that the source can be updated, includes done and websites generated in a consistent and easy procedure for a team of people to use.  
  
Many thanks for providing this tool - my team is very excited it exists because it brings to markdown a feature we liked in DITA and thought we had lost.  

Thanks for being so gentle with a newbie like me.  Jekyll makes websites fun and easy for more than just keen programmers.
